components/nexus-core/src/main/java/org/sonatype/nexus/proxy/item/ChecksumReconciler.java

```diff
@@ -91,30 +97,79 @@ protected void processFileItem(final WalkerContext ctx, final StorageFileItem it
    * Checks the checksum cached in the item's attributes to see if it needs reconciling with the stored content.
    */
   void reconcileItemChecksum(final Repository repo, final StorageFileItem item) throws Exception {
-    final String attributeChecksum = item.getRepositoryItemAttributes().get(DigestCalculatingInspector.DIGEST_SHA1_KEY);
-    if (attributeChecksum != null) {
-
-      message.reset();
-      // look for checksums affected by the link persister pre-fetching the first 8 bytes (see NEXUS-8178)
-      try (final InputStream is = new DigestInputStream(item.getContentLocator().getContent(), message)) {
-        final byte[] buf = new byte[8];
-        ByteStreams.read(is, buf, 0, 8);
-        message.update(buf);
-        ByteStreams.copy(is, ByteStreams.nullOutputStream());
+    final String itemPath = item.getPath();
+    if (!itemPath.endsWith(".sha1") && !itemPath.endsWith(".md5")) {
+      final String attributeSHA1 = item.getRepositoryItemAttributes().get(DigestCalculatingInspector.DIGEST_SHA1_KEY);
+      if (attributeSHA1 != null) {
+
+        sha1.reset();
+        md5.reset();
+
+        final InputStream content = item.getContentLocator().getContent();
```

Comment:
sure, or I could just put it back inside the same line as the wrapping streams (which is effectively the same bytecode-wise, but might make people happier)